### PR TITLE
buildkite: Fix uploading artifacts when e2e test fails

### PR DESCRIPTION
**Pull request by peternose**
Problem: Failed tests didn't have artifacts.

### CODE REVIEW

**Comment by peternose** [comment]
@gw0 Quick fix. Open to better ideas :)

File: .changelog/5382.trivial.md
Comment (kostko) [inline .changelog/5382.trivial.md:1]: Content seems unrelated? :)
